src/CMakeLists.txt

```
@@ -56,12 +56,35 @@ if(BUILD_POLYHEDRAL_GRAVITY_PYTHON_INTERFACE)
 
     target_include_directories(polyhedral_gravity PUBLIC
             "${CMAKE_CURRENT_SOURCE_DIR}"
-            )
+    )
 
     target_link_libraries(polyhedral_gravity PUBLIC
             spdlog::spdlog
             tetgen
             xsimd
             Thrust
-            )
-endif()
\ No newline at end of file
+    )
+endif ()
+
+#####################################################
+# Building the Polyhedral Time Measurement Executable
+#####################################################
+if (BUILD_POLYHEDRAL_GRAVITY_TIME_MEASUREMENT)
+    message(STATUS "Building the Polyhedral Gravity Executable")
+
+    include(gtest)
+    include(benchmark)
+
+    # Building the standalone Executable
+    add_executable(${PROJECT_NAME}_time kd_time_main.cpp)
+
+    # Link executable with library
+    target_link_libraries(${PROJECT_NAME}_time PUBLIC ${PROJECT_NAME}_lib benchmark)
```

```suggestion
    target_link_libraries(${PROJECT_NAME}_time PUBLIC ${PROJECT_NAME}_lib benchmark::benchmark)
```

Otherwise, it does not necessarily work if `Google Benchmark` is pre-installed and not setup via `FetchContent`.